core/util_test.go

```diff
@@ -798,6 +806,8 @@ func (d *DrandTestScenario) RunReshare(t *testing.T, c *reshareConfig) (*key.Gro
 			t.Logf("[reshare] Received group!")
 			d.newGroup = finalGroup
 			require.NoError(d.t, key.Save(d.groupPath, d.newGroup, false))
+			// if we got the group from the leader, DKG was a success, and we wait for all to terminate their DKG
+			wg.Wait()
```

Comment:
Should this be above the for loop? Or we just need the leader to finish?

Reply:
No, because until we've received the final group from the leader, we haven't finished the resharing. Moving this above the `for` loop would prevent use from going through the different cases in the select statement. This could be a deferred statement I guess, but I don't want the `Wait` to block us in case of a timeout.